javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java

```diff
@@ -109,28 +110,28 @@ default Type getElementType() {
      * <br/>For <code>int[] a[][],b[],c[][];</code> this is int[][].
      */
     @DerivedProperty
-    default Type getMaximumCommonType() {
-        return calculateMaximumCommonType(this.getVariables().stream().map(v -> v.getType()).collect(Collectors.toList()));
+    default Optional<Type> getMaximumCommonType() {
```

Comment:
This is the main change: there is not always a maximum common type, don't fail hard on it.